pkg/window/strategy/sliding/sliding.go

```diff
@@ -24,51 +24,109 @@ package sliding
-	// successive windows are phased out by this duration.
-	Slide time.Duration
-	// entries is the list of active windows that are currently being tracked.
-	// windows are sorted in chronological order with the earliest window at the head of the list.
-	// list.List is implemented as a doubly linked list which allows us to traverse the nodes in
-	// both the directions.
-	// Although the worst case time complexity is O(n), because of the time based ordering and
-	// since the elements are rarely out of order, the amortized complexity works out to be closer to O(1)
-	// Because most of the keys are expected to be associated with the most recent window, we always start
-	// the traversal from the tail of the list for Get and Create Operations. For Remove Operations, since
-	// the earlier windows are expected to be closed before the more recent ones, we start the traversal
-	// from the Head.
-	entries *window.SortedWindowList[window.AlignedKeyedWindower]
-}
-
-var _ window.Windower = (*Sliding)(nil)
-
-// NewSliding returns a Sliding windower
-func NewSliding(length time.Duration, slide time.Duration) *Sliding {
-	return &Sliding{
-		Length:  length,
-		Slide:   slide,
-		entries: window.NewSortedWindowList[window.AlignedKeyedWindower](),
+// slidingWindow TimedWindow implementation for Sliding window.
+type slidingWindow struct {
+	startTime time.Time
+	endTime   time.Time
+	slot      string
+}
+
+func NewSessionWindow(startTime time.Time, endTime time.Time) window.TimedWindow {
+	// TODO: slot should be extracted based on the key
+	// we can accept an interface SlotAssigner
+	// which will assign the slot based on the key
+	// slot := slotAssigner.AssignSlot(message.Key)
+	// for now we are using slot-0
+	slot := "slot-0"
+	return &slidingWindow{
+		startTime: startTime,
+		endTime:   endTime,
+		slot:      slot,
+	}
+}
+
+func (w *slidingWindow) StartTime() time.Time {
+	return w.startTime
+}
+
+func (w *slidingWindow) EndTime() time.Time {
+	return w.endTime
+}
+
+func (w *slidingWindow) Slot() string {
+	return w.slot
+}
+
+func (w *slidingWindow) Keys() []string {
+	return nil
+}
+
+func (w *slidingWindow) Partition() *partition.ID {
+	return &partition.ID{
+		Start: w.startTime,
+		End:   w.endTime,
+		Slot:  w.slot,
 	}
 }
 
-// AssignWindow returns a set of windows that contain the element based on event time
-func (s *Sliding) AssignWindow(eventTime time.Time) []window.AlignedKeyedWindower {
-	windows := make([]window.AlignedKeyedWindower, 0)
+// Merge merges the given window with the current window.
+func (w *slidingWindow) Merge(tw window.TimedWindow) {
+	// never be invokved for Aligned Window
+}
+
+func (w *slidingWindow) Expand(endTime time.Time) {
+	// never be invokved for Aligned Window
+}
+
+// Windower is a implementation of TimedWindower of fixed window, windower is responsible for assigning
+// windows to the incoming messages and closing the windows that are past the watermark.
+type Windower struct {
```

Comment:
I'm not a fan of prefixing type names with package names, it's redundant since the type is already inside that package. It's implicitly understood. For example, I prefer 'sliding.Windower' instead of 'sliding.SlidingWindower'.